command/setup_consul.go

```diff
@@ -0,0 +1,736 @@
+	existingRules, _, _ := s.client.ACL().BindingRuleList("", nil)
+	return slices.ContainsFunc(
+		existingRules,
+		func(r *api.ACLBindingRule) bool { return r.AuthMethod == rule.AuthMethod })
+}
+
+func (s *SetupConsulCommand) createBindingRules(rule *api.ACLBindingRule) error {
+	bindingRule, _, err := s.client.ACL().BindingRuleCreate(rule, nil)
+	if err != nil {
+		return fmt.Errorf("[✘] Could not create Consul binding rule: %w", err)
+	}
+
+	s.Ui.Info(fmt.Sprintf("[✔] Created binding rule for auth method %q.", rule.AuthMethod))
+	s.bindingRuleIDs = append(s.bindingRuleIDs, bindingRule.ID)
+
+	return nil
+}
+
+func (s *SetupConsulCommand) roleExists() bool {
+	existingRoles, _, _ := s.client.ACL().RoleList(nil)
+	return slices.ContainsFunc(
+		existingRoles,
+		func(r *api.ACLRole) bool { return r.Name == consulRoleTasks })
+}
+
+func (s *SetupConsulCommand) createRoleForTasks() error {
+	_, _, err := s.client.ACL().RoleCreate(&api.ACLRole{
+		Name:        consulRoleTasks,
+		Description: "Role for Nomad tasks using workload identities",
+		Policies:    []*api.ACLLink{{Name: consulPolicyName}},
+	}, nil)
+	if err != nil {
+		return fmt.Errorf("[✘] Could not create Consul role: %w", err)
+	}
+
+	s.Ui.Info(fmt.Sprintf("[✔] Created role %q.", consulRoleTasks))
+	return nil
+}
+
+func (s *SetupConsulCommand) policyExists() bool {
+	existingPolicies, _, _ := s.client.ACL().PolicyList(nil)
+	return slices.ContainsFunc(
+		existingPolicies,
+		func(p *api.ACLPolicyListEntry) bool { return p.Name == consulPolicyName })
+}
+
+func (s *SetupConsulCommand) createPolicy() error {
+	policy, _, err := s.client.ACL().PolicyCreate(&api.ACLPolicy{
+		Name:  consulPolicyName,
+		Rules: string(consulPolicyBody),
+	}, nil)
+	if err != nil {
+		return fmt.Errorf("[✘] Could not create Consul policy: %w", err)
+	}
+
+	s.Ui.Info(fmt.Sprintf("[✔] Created policy %q.", consulPolicyName))
+	s.policyID = policy.ID
+
+	return nil
+}
+
+// askQuestion asks question to user until they provide a valid response.
+func (s *SetupConsulCommand) askQuestion(question string) bool {
+	for {
+		answer, err := s.Ui.Ask(s.Colorize().Color(fmt.Sprintf("[?] %s", question)))
+		if err != nil {
+			if err.Error() != "interrupted" {
+				s.Ui.Output(err.Error())
+				os.Exit(1)
+			}
+			os.Exit(0)
+		}
+
+		switch strings.TrimSpace(strings.ToLower(answer)) {
+		case "", "y", "yes":
+			return true
+		case "n", "no":
+			return false
+		default:
+			s.Ui.Output(fmt.Sprintf(`%q is not a valid response, please answer "yes" or "no".`, answer))
+			continue
+		}
+	}
+}
+
+func (s *SetupConsulCommand) handleNo() {
+	s.Ui.Warn(`
+By answering "no" to any of these questions, you are risking an incorrect Consul
+cluster configuration. Nomad workloads with Workload Identity will not be able
+to authenticate unless you create missing configuration yourself.
+`)
+
+	if !s.autoYes && s.askQuestion("Remove everything this command creates? [Y/n]") {
```

Comment:
Can we list the things that will be deleted here? It could be a static list like:

```
[?] Remove everything this command creates? [Y/n]

The following items will be deleted if they exist:
  * Auth methods "nomad-services" and "nomad-tasks".
  * Binding rules associated with these methods.
  * Role named "role-nomad-task"
  * Policy named "nomad-workloads".
```

Or a dynamic list, where we first collect what needs to be deleted and output exactly what will be removed.

```
[?] Remove everything this command creates? [Y/n]

The following items will be deleted:
  * Auth methods "nomad-services" and "nomad-tasks".
  * Binding rules "ABC" and "XYZ".
  * Policy named "nomad-workloads.
```

Reply:
implemented in [599150e](https://github.com/hashicorp/nomad/pull/18820/commits/599150edb88dd997338d83c51c719014768a12fc)